country_plan/models.py

```diff
@@ -0,0 +1,105 @@
+from django.db import models
+from django.contrib.auth.models import User
+from django.utils import timezone
+from django.utils.translation import gettext_lazy as _
+
+from api.models import Country
+
+
+def file_upload_to(instance, filename):
+    date_str = timezone.now().strftime('%Y-%m-%d-%H-%M-%S')
+    return f'country-plan-exel-files/{date_str}/{filename}'
+
+
+def pdf_upload_to(instance, filename):
+    date_str = timezone.now().strftime('%Y-%m-%d-%H-%M-%S')
+    return f'country-plan-pdf/{date_str}/{filename}'
+
+
+class CountryPlanBase(models.Model):
+    created_by = models.ForeignKey(
+        User,
+        verbose_name=_('Created By'),
+        blank=True,
+        null=True,
+        related_name='data_import_created_by',
+        on_delete=models.SET_NULL,
+    )
+    created_at = models.DateTimeField(verbose_name=_('Created at'), auto_now_add=True)
+    updated_by = models.ForeignKey(
+        User,
+        verbose_name=_('Updated by'),
+        blank=True,
+        null=True,
+        related_name='data_import_updated_by',
+        on_delete=models.SET_NULL
+    )
+    updated_at = models.DateTimeField(verbose_name=_('Updated at'), auto_now=True)
+
+
+class DataImport(CountryPlanBase):
+    file = models.FileField(verbose_name=_('EXEL file'), upload_to=file_upload_to)
+
+    def __str__(self):
+        return self.file.name
+
+
+class CountryPlan(CountryPlanBase):
+    internal_plan_file = models.FileField(verbose_name=_('Internal Plan'), upload_to=pdf_upload_to)
+    public_plan_file = models.FileField(verbose_name=_('Country Plan'), upload_to=pdf_upload_to)
+    country = models.ForeignKey(Country, on_delete=models.SET_NULL, null=True, blank=True)
```

Comment:
Can `country` be null?

Reply:
@k9845 country is set null=True to prevent data loss for countryplan if country is deleted.